src/main/java/org/opentripplanner/middleware/connecteddataplatform/ConnectedDataManager.java

```diff
@@ -110,11 +115,18 @@ public static boolean canScheduleUploads() {
 
     public static void scheduleTripHistoryUploadJob() {
         if (canScheduleUploads()) {
-            LOG.info("Scheduling trip history upload for every {} minute(s)",
-                CONNECTED_DATA_PLATFORM_TRIP_HISTORY_UPLOAD_JOB_FREQUENCY_IN_MINUTES);
+            LOG.info("Scheduling trip history upload for every {} minute(s) starting at {}",
+                CONNECTED_DATA_PLATFORM_TRIP_HISTORY_UPLOAD_JOB_FREQUENCY_IN_MINUTES,
+                CONNECTED_DATA_PLATFORM_TRIP_HISTORY_UPLOAD_START_TIME);
+
+            var now = DateTimeUtils.nowAsZonedDateTime(DateTimeUtils.getOtpZoneId());
+            var timeOfDay = LocalTime.parse(CONNECTED_DATA_PLATFORM_TRIP_HISTORY_UPLOAD_START_TIME);
+            var startAt = DateTimeUtils.getNextTimeFrom(timeOfDay, now);
+            long initialDelayMinutes = Duration.between(now, startAt).toMinutes();
```

Comment:
One detail that I overlooked is that the configured start time should only apply if the uploads are daily (`CONNECTED_DATA_PLATFORM_REPORTING_INTERVAL = "daily"`). If the reporting interval is hourly, the job should still start immediately.

Reply:
Extract a method to compute the initial delay (lines 122 to 125), and place it in `DateTimeUtils` or `Scheduler`, and add a unit test for that method.